### PR TITLE
Feature/invoices actions

### PR DESCRIPTION
**Pull request by orzechdev**
I want to merge this change because it adds send and generate invoice actions.

**PR intended to be tested with API branch:** 

https://github.com/mirumee/saleor/pull/5732

### Screenshots

Zrzut ekranu 2020-07-2 o 11 21 37
Zrzut ekranu 2020-07-1 o 12 15 06

### Pull Request Checklist

1. [ ] All visible strings are translated with proper context.
1. [ ] All data-formatting is locale-aware (dates, numbers, and so on).
1. [ ] Translated strings are extracted.
1. [ ] Number of API calls is optimized.
1. [ ] The changes are tested.
1. [ ] Type definitions are up to date.
1. [ ] Changes are mentioned in the changelog.

### CODE REVIEW

**Comment by orzechdev** [comment]
Task linked: [CU-6a5w7q Trigger creation of invoice](https://app.clickup.com/t/6a5w7q)

**Comment by orzechdev** [comment]
Task linked: [CU-6a5w7x Trigger sending an email with invoice](https://app.clickup.com/t/6a5w7x)

File: src/orders/components/OrderInvoiceEmailSendDialog/OrderInvoiceEmailSendDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+            values={{
+              invoiceNumber: <strong>{invoice?.number}</strong>
+            }}
+          />
+        </DialogContentText>
+        {errors.length > 0 && (
+          <>
+            <FormSpacer />
+            {errors.map(err => (
+              <DialogContentText color="error">
```
Comment (krzysztofwolski) [inline src/orders/components/OrderInvoiceEmailSendDialog/OrderInvoiceEmailSendDialog.tsx:58]: Add key

File: src/orders/components/OrderInvoiceEmailSendDialog/OrderInvoiceEmailSendDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+import { FormattedMessage, useIntl } from "react-intl";
+
+export interface OrderInvoiceEmailSendDialogProps {
+  confirmButtonState: ConfirmButtonTransitionState;
+  errors: InvoiceErrorFragment[];
+  open: boolean;
+  invoice: InvoiceFragment;
+  onClose: () => void;
+  onSend: () => void;
+}
```
Comment (dominik-zeglen) [inline src/orders/components/OrderInvoiceEmailSendDialog/OrderInvoiceEmailSendDialog.tsx:25]: Let's keep things tidy

```suggestion
export interface OrderInvoiceEmailSendDialogProps extends DialogProps {
  confirmButtonState: ConfirmButtonTransitionState;
  errors: InvoiceErrorFragment[];
  invoice: InvoiceFragment;
  onSend: () => void;
}
```

File: src/orders/components/OrderInvoiceList/OrderInvoiceList.tsx [hunk 3]
```diff
@@ -37,9 +37,9 @@ const useStyles = makeStyles(
 
 interface OrderInvoiceListProps {
   invoices: InvoiceFragment[];
-  onInvoiceGenerate?: () => void;
-  onInvoiceClick?: (invoice: InvoiceFragment) => void;
-  onInvoiceSend?: (invoice: InvoiceFragment) => void;
+  onInvoiceGenerate: () => void;
+  onInvoiceClick: (invoice: InvoiceFragment) => void;
+  onInvoiceSend: (invoice: InvoiceFragment) => void;
```
Comment (dominik-zeglen) [inline src/orders/components/OrderInvoiceList/OrderInvoiceList.tsx:42]: Everywhere else we use ids so let's keep it that way

```suggestion
  onInvoiceClick: (invoiceId: string) => void;
  onInvoiceSend: (invoiceId: string) => void;
```

File: src/orders/components/OrderInvoiceEmailSendDialog/OrderInvoiceEmailSendDialog.tsx (hunk 1 above)
Comment (orzechdev) [reply src/orders/components/OrderInvoiceEmailSendDialog/OrderInvoiceEmailSendDialog.tsx]: fixed

File: src/orders/components/OrderDetailsPage/OrderDetailsPage.tsx [hunk 4]
```diff
@@ -65,9 +65,11 @@ export interface OrderDetailsPageProps extends UserPermissionProps {
   onNoteAdd(data: HistoryFormData);
   onProfileView();
   onInvoiceClick(invoice: InvoiceFragment);
+  onInvoiceGenerate();
+  onInvoiceSend(invoice: InvoiceFragment);
```
[the remark this answers is not in the document]
Comment (orzechdev) [reply src/orders/components/OrderDetailsPage/OrderDetailsPage.tsx]: fixed

File: src/orders/components/OrderInvoiceEmailSendDialog/OrderInvoiceEmailSendDialog.tsx (hunk 2 above)
Comment (orzechdev) [reply src/orders/components/OrderInvoiceEmailSendDialog/OrderInvoiceEmailSendDialog.tsx]: fixed

File: src/orders/components/OrderInvoiceList/OrderInvoiceList.tsx [hunk 5]
```diff
@@ -67,10 +71,12 @@ const OrderInvoiceList: React.FC<OrderInvoiceListProps> = props => {
...
       />
-      <CardContent className={invoices?.length && classes.cardContentTable}>
-        {!invoices ? (
+      <CardContent
+        className={generatedInvoices?.length && classes.cardContentTable}
+      >
+        {!generatedInvoices ? (
           <Skeleton />
-        ) : !invoices?.length ? (
+        ) : !generatedInvoices?.length ? (
```
Comment (dominik-zeglen) [inline src/orders/components/OrderInvoiceList/OrderInvoiceList.tsx:79]: Please add stories covering `undefined`, `[]` and non-empty state

Comment (orzechdev) [reply src/orders/components/OrderInvoiceList/OrderInvoiceList.tsx:79]: added

**Pull request merged**